### ISSUE TITLE
Android O IllegalStateException in PushGcmIntentService in android 8.0

### ISSUE THREAD

**Issue by Prabha10cs** [Pushwoosh/pushwoosh-android-sdk#82]
how to solve this problem
E/AndroidRuntime: FATAL EXCEPTION: main
                                                                    Process: com.gcm.notification, PID: 3021
                                                                    java.lang.RuntimeException: Unable to start receiver com.google.android.gcm.GCMBroadcastReceiver: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x1000010 pkg=com.gcm.notification cmp=com.gcm.notification/.GCMIntentService (has extras) }: app is in background uid UidRecord{b85959f u0a84 RCVR idle procs:1 seq(0,0,0)}
                                                                        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3259)
                                                                        at android.app.ActivityThread.-wrap17(Unknown Source:0)
                                                                        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1677)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                        at android.os.Looper.loop(Looper.java:164)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:6541)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                                     Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x1000010 pkg=com.gcm.notification cmp=com.gcm.notification/.GCMIntentService (has extras) }: app is in background uid UidRecord{b85959f u0a84 RCVR idle procs:1 seq(0,0,0)}
                                                                        at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1505)
                                                                        at android.app.ContextImpl.startService(ContextImpl.java:1461)
                                                                        at android.content.ContextWrapper.startService(ContextWrapper.java:644)
                                                                        at android.content.ContextWrapper.startService(ContextWrapper.java:644)
                                                                        at com.google.android.gcm.GCMBaseIntentService.runIntentInService(GCMBaseIntentService.java:282)
                                                                        at com.google.android.gcm.GCMBroadcastReceiver.onReceive(GCMBroadcastReceiver.java:55)
                                                                        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3252)
                                                                        at android.app.ActivityThread.-wrap17(Unknown Source:0) 
                                                                        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1677) 
                                                                        at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                                        at android.os.Looper.loop(Looper.java:164) 
                                                                        at android.app.ActivityThread.main(ActivityThread.java:6541) 
                                                                        at java.lang.reflect.Method.invoke(Native Method) 
                                                                        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
how to solve this problem

**Comment by wfhm** [comment]
Hi,

Could you please specify Pushwoosh SDK version used in your project? Also, which Android OS version is affected?

**Comment by Prabha10cs** [comment]
@wfhm 
android OS version 8.0 and 8.1

**Comment by wfhm** [comment]
@Prabha10cs Thank you! And what about Pushwoosh SDK version?

**Comment by Vichy97** [comment]
I am also running into issues. I think it has to do with the new way android handles background services in android 8.0

**Comment by wfhm** [comment]
@Vichy97  could you please specify Pushwoosh SDK version you use?

**Comment by Vichy97** [comment]
@wfhm whatever version was right before gradle 4 was required. I havent tested on the latest because I cant get the latest version to build

**Comment by wfhm** [comment]
@Vichy97 

The initial issue is already fixed in 5.1 versions of our SDK. With the latest release, you can use Pushwoosh SDK in your project even without using gradle 4. To do that, you should add the following edits to your build.gradle:

```
allprojects {
    repositories {
        jcenter ()
        google()
    }
}
```

```
implementation 'com.pushwoosh.java7:pushwoosh:5.+'
implementation 'com.android.support:support-compat:27.+'
```
